src/core/DY.Core.Bytes.Type.fst

```diff
@@ -52,6 +52,9 @@ type bytes =
   | KemPub: sk:bytes -> bytes
   | KemEncap: pk:bytes -> ss:bytes -> bytes
   | KemSecretShared: ss:bytes -> bytes
+
+  | Mac: key:bytes -> msg:bytes -> bytes
```

Comment:
Should a MAC also include a nonce, as with both encryption methods and signatures?

Reply:
It is true that in all generality MAC algorithms may be randomized, and we could model that with a nonce. However I think every "real-world" MAC algorithm (HMAC, Poly1305, CMAC etc) are deterministic hence I would not put a nonce there.

Reply:
Seems reasonable for now, at least (and not hard to  add if that changes, or we want to model some protocol that specifies using an unusual MAC).